SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -171,7 +171,33 @@ void ComponentReader::ApplySchemaObject(Schema_Object* ComponentObject, UObject&
 	bool bIsClient = NetDriver->GetNetMode() == NM_Client;
 	bool bEventTracerEnabled = EventTracer != nullptr;
 
-	FSpatialConditionMapFilter ConditionMap(&Channel, bIsClient);
+	// RemoteRole and bRepPhysics are used to construct the condition map. We must check to see if these
+	// properties have just been received, before we construct the condition map for receiving everything else.
+	ENetRole ActorRole = Channel.Actor->Role;
+	bool bRepPhysics = Channel.Actor->GetReplicatedMovement().bRepPhysics;
+	if (EnumHasAnyFlags(Replicator->RepLayout->GetFlags(), ERepLayoutFlags::IsActor))
+	{
+		for (uint32 FieldId : UpdatedIds)
+		{
+			int32 CmdIndex = BaseHandleToCmdIndex[FieldId - 1].CmdIndex;
+			const FRepLayoutCmd& Cmd = Cmds[CmdIndex];
+
+			if (UNLIKELY((int32)AActor::ENetFields_Private::RemoteRole == Cmd.ParentIndex))
+			{
+				ActorRole = (ENetRole)Schema_IndexUint32(ComponentObject, FieldId, 0);
+				if (ActorRole == ROLE_AutonomousProxy && !bIsAuthServer && !bAutonomousProxy)
+				{
+					ActorRole = ROLE_SimulatedProxy;
```

Comment:
Can we add a comment here talking about the race that exists where the client might check out the actor before they know they're auth on it, and where that gets fixed up? (in ActorSystem::HandleActorAuthority)